rust/src/lib/nm/settings/sriov.rs

```diff
@@ -27,6 +27,13 @@ pub(crate) fn gen_nm_sriov_setting(
         nm_sriov_set.total_vfs = Some(v);
     }
 
+    if let Some(autoprobe) = sriov_conf.drivers_autoprobe {
+        nm_sriov_set.autoprobe_drivers = Some(autoprobe);
+    } else {
+        // if unset, always force to true so that nm restores the kernel default
+        nm_sriov_set.autoprobe_drivers = Some(true);
```

Comment:
You are overriding current setting even when desired state not mentioned so.

For example, assuming pre-apply has `autoprobe_drivers: false`, when you applying desired state changing total vfs count only, your patch will set `autoprobe_drivers: true` explicitly.

Reply:
If you want to default `autoprobe_drivers` to be true if not mentioned in desire state and current state, please use the same workflow as https://github.com/nmstate/nmstate/pull/2665